File: notused/gen.py
MODE  = 'clash'
MODE = 'quan x'
urls_proxy = []
urls_direct = []

DIRECT = ""
# DIRECT
## Microsoft
DIRECT += "microsoft.com microsoftonline.com sharepoint.com office.com office.net "
## Crusaders Quest
DIRECT += "cq.hangame.com cq-pvp.hangame.com cq-cha.hangame.com gslb-gamebase.nhncloudservice.com " # ios
DIRECT += ""
urls_direct.extend(DIRECT.split(' '))

PROXY = "amp-api-edge.apps.apple.com push.apple.com inappcheck.itunes.apple.com app-measurement.com nexoncdn.co.kr nexon.com nexon.io "
# PROXY
urls_proxy.extend(PROXY.split(' '))

urls_proxy = set(urls_proxy)
urls_direct = set(urls_direct)




















# if mode == 'clash':
#     res = requests.get("https://ghelper.me/clash/4ff905a0af778432e7dd12dfe21788cb")
#     proxy = res.text.split('rules:')[0]
#     print(proxy)
#     print('rules:')
#     for token in urls_proxy:
#         print('- DOMAIN-SUFFIX,{},Ghelper'.format(token))
#     for token in urls_direct:
#         print('- DOMAIN-SUFFIX,{},DIRECT'.format(token))
#     s = '# default\n- DOMAIN-SUFFIX,ip-api.com,DIRECT\n- DOMAIN-SUFFIX,ipip.net,DIRECT\n- DOMAIN-SUFFIX,ip138.com,DIRECT\n- DOMAIN-SUFFIX,stunnel.vip,DIRECT\n- DOMAIN-SUFFIX,gotochinatown.net,DIRECT\n- DOMAIN-SUFFIX,ghelper.net,DIRECT\n- DOMAIN-SUFFIX,ghelper.me,DIRECT\n- DOMAIN-SUFFIX,ghelper.xyz,DIRECT\n- DOMAIN-SUFFIX,ghelper.org,DIRECT\n- DOMAIN-SUFFIX,fastapi.me,DIRECT\n- DOMAIN-SUFFIX,vps315.com,DIRECT\n- DOMAIN-SUFFIX,copyplay.net,DIRECT\n- DOMAIN-SUFFIX,pickdown.net,DIRECT\n- DOMAIN-KEYWORD,google,Ghelper\n- DOMAIN-KEYWORD,youtube,Ghelper\n- DOMAIN-SUFFIX,ggpht.com,Ghelper\n- DOMAIN-SUFFIX,gmail.com,Ghelper\n- DOMAIN-SUFFIX,gvt2.com,Ghelper\n- DOMAIN-SUFFIX,gvt3.com,Ghelper\n- DOMAIN-SUFFIX,chrome.com,Ghelper\n- DOMAIN-SUFFIX,wikipedia.org,Ghelper\n- DOMAIN-SUFFIX,wikimedia.org,Ghelper\n- DOMAIN-SUFFIX,appspot.com,Ghelper\n- DOMAIN-SUFFIX,android.com,Ghelper\n- DOMAIN-SUFFIX,github.com,Ghelper\n- DOMAIN-SUFFIX,gitbook.com,Ghelper\n- DOMAIN-SUFFIX,kaggle.com,Ghelper\n- DOMAIN-SUFFIX,arxiv.org,Ghelper\n- DOMAIN-SUFFIX,wiktionary.org,Ghelper\n- DOMAIN-SUFFIX,blogger.com,Ghelper\n- DOMAIN-SUFFIX,youtu.be,Ghelper\n- DOMAIN-SUFFIX,ytimg.com,Ghelper\n- DOMAIN-SUFFIX,youtube.com,Ghelper\n- DOMAIN-SUFFIX,instagram.com,Ghelper\n- DOMAIN-SUFFIX,twitter.com,Ghelper\n- DOMAIN-SUFFIX,t.co,Ghelper\n- DOMAIN-SUFFIX,facebook.com,Ghelper\n- DOMAIN-SUFFIX,telegram.org,Ghelper\n- IP-CIDR,192.168.0.0/16,DIRECT\n- IP-CIDR,10.0.0.0/8,DIRECT\n- IP-CIDR,172.16.0.0/12,DIRECT\n- IP-CIDR,172.23.0.0/12,DIRECT\n- IP-CIDR,127.0.0.0/8,DIRECT\n- IP-CIDR,100.64.0.0/10,DIRECT\n- IP-CIDR6,::1/128,DIRECT\n- IP-CIDR6,fc00::/7,DIRECT\n- IP-CIDR6,fe80::/10,DIRECT\n- IP-CIDR6,fd00::/8,DIRECT\n- GEOIP,CN,DIRECT\n- MATCH,DIRECT'
#     print(s)

if MODE == "quan x":
    for token in urls_proxy:
        print('host-suffix,{},proxy'.format(token))
    for token in urls_direct:
        print('host-suffix,{},direct'.format(token))
    # s = 'ip-cidr, 10.0.0.0/8, direct\nip-cidr, 127.0.0.0/8, direct\nip-cidr, 172.16.0.0/12, direct\nip-cidr, 192.168.0.0/16, direct\nip-cidr, 224.0.0.0/24, direct\ngeoip, cn, direct\nfinal, direct'
if MODE == "nekoray":
    pass

File: cmd/gen.go
package main

import (
	"fmt"
	"os"
	"strings"
)

const (
	SHADOWROCKET_PREFIX = `
[General]
bypass-system = true
skip-proxy = 192.168.0.0/16, 10.0.0.0/8, 172.16.0.0/12, localhost, *.local, captive.apple.com
tun-excluded-routes = 10.0.0.0/8, 100.64.0.0/10, 127.0.0.0/8, 169.254.0.0/16, 172.16.0.0/12, 192.0.0.0/24, 192.0.2.0/24, 192.88.99.0/24, 192.168.0.0/16, 198.51.100.0/24, 203.0.113.0/24, 224.0.0.0/4, 255.255.255.255/32, 239.255.255.250/32
dns-server = system
fallback-dns-server = system
ipv6 = true
prefer-ipv6 = false
dns-fallback-system = false
dns-direct-system = false
icmp-auto-reply = true
always-reject-url-rewrite = false
private-ip-answer = true
# direct domain fail to resolve use proxy rule
dns-direct-fallback-proxy = true
# The fallback behavior when UDP traffic matches a policy that doesn't support the UDP relay. Possible values: DIRECT, REJECT.
udp-policy-not-supported-behaviour = REJECT

[Rule]
`
	SHADOWROCKET_SUFFIX = `
# LAN
IP-CIDR,192.168.0.0/16,DIRECT
IP-CIDR,10.0.0.0/8,DIRECT
IP-CIDR,172.16.0.0/12,DIRECT
IP-CIDR,127.0.0.0/8,DIRECT
# China
GEOIP,CN,DIRECT
# Final
FINAL,PROXY

[Host]
localhost = 127.0.0.1

[URL Rewrite]
^https?://(www.)?g.cn https://www.google.com 302
^https?://(www.)?google.cn https://www.google.com 302
`
)

func main() {
	MODES := []string{
		"clash",
		"quan x",
		"matsuri",
		"surge",
		"shadowrocket",
	}

	var domainProxy, domainDirect, domainReject []string

	// PROXY
	PROXY := "amp-api-edge.apps.apple.com push.apple.com inappcheck.itunes.apple.com nexoncdn.co.kr nexon.com nexon.io "
	// Google
	{
		PROXY += "googleapis.com "
	}
	// Microsoft
	{
		PROXY += "bing.com "
		PROXY += "windows.com windows.net office.com microsoft.com live.com "
		PROXY += "onenote.com contentsync.onenote.com hierarchyapi.onenote.com "
		PROXY += "microsoftonline.com office.net " // 原来是direct
		PROXY += "sharepoint.com 1drv.com "        // onedrive
		PROXY += "googleapis.cn "
	}
	// Apple
	{
		PROXY += "app.adjust.com "
	}
	// Crusaders Quest
	// {
	// 	PROXY += "cq.hangame.com "
	// 	PROXY += "nhn.com gslb-gamebase.nhncloudservice.com toast.com "
	// }
	// LinkedIn
	{
		PROXY += "linkedin.com "
	}
	// Leetcode
	{
		PROXY += "leetcode.cn "
	}

	// DIRECT
	DIRECT := ""
	// cn
	{
		DIRECT += "*baidu* baidu.com qq.com zhihu.com dcarstatic.com byteimg.com 163.com csdn.net qcloudimg.com tencent.com bilibili.com "
	}
	// Common
	{
		DIRECT += "aliyuncs.com "
	}
	// alist
	{
		DIRECT += "alist.r1cebucket.top "
		DIRECT += "alist-proxy.r1cebucket.top "
	}
	// Microsoft
	{
		// DIRECT += ""
	}
	// Crusaders Quest
	// {
	// 	DIRECT += "nhnst.com "
	// 	DIRECT += "cq-pvp.hangame.com " // 匹配队友，对手（需要直连）
	// 	DIRECT += "cq-cha.hangame.com " // 进入匹配服务器（需要直连）
	// 	DIRECT += "toastoven.net "
	// 	// DIRECT += "cru.cdn.toastoven.net adam.gslb.toastoven.net api-iaptacc.gslb.toastoven.net "
	// 	DIRECT += "unity3d.com "
	// }
	// Steam
	{
		DIRECT += "akamaihd.net "
	}
	// WeTab
	{
		DIRECT += "wetab.link "
	}

	// REJECT
	REJECT := ""
	// analytic
	{
		REJECT += "app-measurement.com appsflyer.com google-analytics.com openinstall.io "

	}
	// ad
	{
		REJECT += "amazon-adsystem.com doubleclick.net rubiconproject.com adservice.google.com wwads.cn "
	}
	// Baidu
	{
		REJECT += "tieba-ares.cdn.bcebos.com "
	}

	domainProxy = strings.Split(PROXY, " ")
	domainDirect = strings.Split(DIRECT, " ")
	domainReject = strings.Split(REJECT, " ")

	for _, MODE := range MODES {
		SaveConfig(domainReject, domainProxy, domainDirect, MODE)
	}
}

func SaveConfig(domainReject, domainProxy, domainDirect []string, MODE string) {
	var rule string
	switch MODE {
	case "quan x":
		conf, err := os.Create("./rules/quan_x.conf")
		if err != nil {
			fmt.Println(err.Error())
			return
		}
		defer conf.Close()

		for _, domain := range domainReject[:len(domainReject)-1] {
			rule = fmt.Sprintf("HOST-SUFFIX,%s,REJECT\n", domain)
			conf.Write([]byte(rule))
		}
		for _, domain := range domainProxy[:len(domainProxy)-1] {
			rule = fmt.Sprintf("HOST-SUFFIX,%s,PROXY\n", domain)
			conf.Write([]byte(rule))
		}
		for _, domain := range domainDirect[:len(domainDirect)-1] {
			rule = fmt.Sprintf("HOST-SUFFIX,%s,DIRECT\n", domain)
			conf.Write([]byte(rule))
		}
	case "matsuri":
		confReject, err := os.Create("./rules/matsuri_reject.conf")
		if err != nil {
			fmt.Println(err.Error())
			return
		}
		confProxy, err := os.Create("./rules/matsuri_proxy.conf")
		if err != nil {
			fmt.Println(err.Error())
			return
		}
		confDirect, err := os.Create("./rules/matsuri_direct.conf")
		if err != nil {
			fmt.Println(err.Error())
			return
		}
		defer confReject.Close()
		defer confProxy.Close()
		defer confDirect.Close()

		for _, domain := range domainReject[:len(domainReject)-1] {
			rule = fmt.Sprintf("domain:%s\n", domain)
			confReject.Write([]byte(rule))
		}
		for _, domain := range domainProxy[:len(domainProxy)-1] {
			rule = fmt.Sprintf("domain:%s\n", domain)
			confProxy.Write([]byte(rule))
		}
		confDirect.Write([]byte("geosite:cn\n"))
		for _, domain := range domainDirect[:len(domainDirect)-1] {
			rule = fmt.Sprintf("domain:%s\n", domain)
			confDirect.Write([]byte(rule))
		}
	case "clash":
		confReject, err := os.Create("./rules/clash_reject.conf")
		if err != nil {
			fmt.Println(err.Error())
			return
		}
		confProxy, err := os.Create("./rules/clash_proxy.conf")
		if err != nil {
			fmt.Println(err.Error())
			return
		}
		confDirect, err := os.Create("./rules/clash_direct.conf")
		if err != nil {
			fmt.Println(err.Error())
			return
		}
		defer confReject.Close()
		defer confProxy.Close()
		defer confDirect.Close()

		confReject.Write([]byte("payload:\n"))
		confProxy.Write([]byte("payload:\n"))
		confDirect.Write([]byte("payload:\n"))

		for _, domain := range domainReject[:len(domainReject)-1] {
			rule = fmt.Sprintf("  - '+.%s'\n", domain)
			confReject.Write([]byte(rule))
		}
		for _, domain := range domainProxy[:len(domainProxy)-1] {
			rule = fmt.Sprintf("  - '+.%s'\n", domain)
			confProxy.Write([]byte(rule))
		}
		for _, domain := range domainDirect[:len(domainDirect)-1] {
			rule = fmt.Sprintf("  - '+.%s'\n", domain)
			confDirect.Write([]byte(rule))
		}

		conf, err := os.Create("./rules/clash.conf")
		if err != nil {
			fmt.Println(err.Error())
			return
		}
		defer conf.Close()

		for _, domain := range domainReject[:len(domainReject)-1] {
			rule = fmt.Sprintf("  - DOMAIN-SUFFIX,%s,REJECT\n", domain)
			conf.Write([]byte(rule))
		}
		for _, domain := range domainProxy[:len(domainProxy)-1] {
			rule = fmt.Sprintf("  - DOMAIN-SUFFIX,%s,PROXY\n", domain)
			conf.Write([]byte(rule))
		}
		for _, domain := range domainDirect[:len(domainDirect)-1] {
			rule = fmt.Sprintf("  - DOMAIN-SUFFIX,%s,DIRECT\n", domain)
			conf.Write([]byte(rule))
		}
	case "surge":
		confReject, err := os.Create("./rules/surge_reject.conf")
		if err != nil {
			fmt.Println(err.Error())
			return
		}
		confProxy, err := os.Create("./rules/surge_proxy.conf")
		if err != nil {
			fmt.Println(err.Error())
			return
		}
		confDirect, err := os.Create("./rules/surge_direct.conf")
		if err != nil {
			fmt.Println(err.Error())
			return
		}
		defer confReject.Close()
		defer confProxy.Close()
		defer confDirect.Close()

		for _, domain := range domainReject[:len(domainReject)-1] {
			rule = fmt.Sprintf("DOMAIN-SUFFIX,%s\n", domain)
			confReject.Write([]byte(rule))
		}
		for _, domain := range domainProxy[:len(domainProxy)-1] {
			rule = fmt.Sprintf("DOMAIN-SUFFIX,%s\n", domain)
			confProxy.Write([]byte(rule))
		}
		for _, domain := range domainDirect[:len(domainDirect)-1] {
			rule = fmt.Sprintf("DOMAIN-SUFFIX,%s\n", domain)
			confDirect.Write([]byte(rule))
		}
	case "shadowrocket":
		conf, err := os.Create("./rules/shadowrocket.conf")
		if err != nil {
			fmt.Println(err.Error())
			return
		}
		defer conf.Close()

		conf.Write([]byte(SHADOWROCKET_PREFIX))

		for _, domain := range domainReject[:len(domainReject)-1] {
			rule = fmt.Sprintf("DOMAIN-SUFFIX,%s,REJECT\n", domain)
			conf.Write([]byte(rule))
		}
		for _, domain := range domainProxy[:len(domainProxy)-1] {
			rule = fmt.Sprintf("DOMAIN-SUFFIX,%s,PROXY\n", domain)
			conf.Write([]byte(rule))
		}
		for _, domain := range domainDirect[:len(domainDirect)-1] {
			rule = fmt.Sprintf("DOMAIN-SUFFIX,%s,DIRECT\n", domain)
			conf.Write([]byte(rule))
		}

		conf.Write([]byte(SHADOWROCKET_SUFFIX))
	default:
		fmt.Println("no such mode:", MODE)
	}
}
